swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt

```diff
@@ -0,0 +1,198 @@
+import com.netflix.spinnaker.swabbie.repository.TaskTrackingRepository
+import com.netflix.spinnaker.swabbie.repository.ResourceTrackingRepository
+import com.netflix.spinnaker.swabbie.repository.ResourceStateRepository
+import com.netflix.spinnaker.swabbie.repository.ResourceUseTrackingRepository
+import com.netflix.spinnaker.swabbie.rules.RulesEngine
+import com.netflix.spinnaker.swabbie.utils.ApplicationUtils
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
+import java.time.Clock
+import kotlin.system.measureTimeMillis
+
+@Component
+class AmazonLaunchConfigurationHandler(
+  registry: Registry,
+  clock: Clock,
+  notifier: Notifier,
+  resourceTrackingRepository: ResourceTrackingRepository,
+  resourceStateRepository: ResourceStateRepository,
+  resourceOwnerResolver: ResourceOwnerResolver<AmazonLaunchConfiguration>,
+  exclusionPolicies: List<ResourceExclusionPolicy>,
+  applicationEventPublisher: ApplicationEventPublisher,
+  swabbieProperties: SwabbieProperties,
+  dynamicConfigService: DynamicConfigService,
+  private val rulesEngine: RulesEngine,
+  private val aws: AWS,
+  private val orcaService: OrcaService,
+  private val applicationUtils: ApplicationUtils,
+  private val taskTrackingRepository: TaskTrackingRepository,
+  resourceUseTrackingRepository: ResourceUseTrackingRepository,
+  notificationQueue: NotificationQueue
+) : AbstractResourceTypeHandler<AmazonLaunchConfiguration>(
+  registry,
+  clock,
+  rulesEngine,
+  resourceTrackingRepository,
+  resourceStateRepository,
+  exclusionPolicies,
+  resourceOwnerResolver,
+  notifier,
+  applicationEventPublisher,
+  resourceUseTrackingRepository,
+  swabbieProperties,
+  dynamicConfigService,
+  notificationQueue
+) {
+  override fun deleteResources(markedResources: List<MarkedResource>, workConfiguration: WorkConfiguration) {
+    val application = applicationUtils.determineApp(markedResources.first().resource)
+    val launchConfigurationNames = markedResources
+      .map {
+        it.resourceId
+      }.toSet()
+
+    orcaService.orchestrate(
+      OrchestrationRequest(
+        application = application,
+        job = listOf(
+          OrcaJob(
+            type = "deleteLaunchConfiguration",
+            context = mutableMapOf(
+              "credentials" to workConfiguration.account.name,
+              "launchConfigurationNames" to launchConfigurationNames,
+              "cloudProvider" to AWS,
+              "region" to workConfiguration.location
+            )
+          )
+        ),
+        description = "Deleting LaunchConfigurations: $launchConfigurationNames"
+      )
+    ).let { taskResponse ->
+      val completeEvent = TaskCompleteEventInfo(
+        action = Action.DELETE,
+        markedResources = markedResources,
+        workConfiguration = workConfiguration,
+        submittedTimeMillis = clock.instant().toEpochMilli()
+      )
+
```

Comment:
Might be worth adding a log message with the resource / orca task id for debugging ?

Reply:
That's a good idea. I would like to stick that in a common place like so it's called for every handler when a task is submitted. I will do that in a seperate PR.